[coverage] VRRP, L2TP, HSRP, RIP, & Radius tests

guedou opened the pull request:
Here are some fixes and tests for VRRP, L2TP, HSRP, RIP, & Radius.

gpotter2 commented:
@guedou Good job with that...
As a note: you may have noticed that on AppVeyor, codecov seems to act a bit weirdly:
```
codecov
'git' is not recognized as an internal or external command,
operable program or batch file.
'git' is not recognized as an internal or external command,
operable program or batch file.
'git' is not recognized as an internal or external command,
operable program or batch file.
'hg' is not recognized as an internal or external command,
operable program or batch file.
'hg' is not recognized as an internal or external command,
operable program or batch file.
'find' is not recognized as an internal or external command,
operable program or batch file.
      _____          _
     / ____|        | |
    | |     ___   __| | ___  ___ _____   __
    | |    / _ \ / _  |/ _ \/ __/ _ \ \ / /
    | |___| (_) | (_| |  __/ (_| (_) \ V /
     \_____\___/ \____|\___|\___\___/ \_/
                                    v2.0.5
==> Detecting CI provider
    AppVeyor Detected
    Error running `git log -1 --pretty=%B`: None
==> Preparing upload
    Error running `cd C:\projects\scapy && git ls-files`: None
    Error running `git ls-files`: None
    Error running `cd C:\projects\scapy && hg locate`: None
    Error running `hg locate`: None
```

guedou commented:
@gpotter2 the error message is present since we introduced codecov on Windows. The git command is not found in the PATH variable. I did not take the time to fix it as codecov works.

p-l- commented:
This PR seems odd: does it include code from other PRs? Anyway it has conflicts.

guedou commented:
Sorry, I messed up the initial `git branch`. It is now rebased and fixed.

FYI, the test was failing because the HSRP layer overrides the IP.dst. Scapy used my local IP address, and the test was failing everywhere else than locally.

p-l- commented on scapy/layers/hsrp.py:
```diff
@@ -67,7 +67,7 @@ class HSRPmd5(Packet):
         ByteField("padding", 0x00),
         XShortField("flags", 0x00),
-        IPField("sourceip", None),
+        IPField("sourceip", "127.0.0.1"),
```
Can you merge #466? That would make the current `None` default value working and more relevant that `"127.0.0.1"`.

guedou replied:
Do you want me to change IPField to SourceIPField here ?

p-l- replied:
Yes, if that's supposed to be a source IP address.

p-l- replied:
Now if you rebase against master, you can change this to `SourceIPField("sourceip", None)`

p-l- replied:
Or I can do that in my PR about `SourceIPField`.

guedou commented:
@p-l- can you restart the AppVeyor build ? It seems that it failed due to a networking issue. Thanks.

guedou commented:
Thanks !
